src/fileio.cpp

```diff
@@ -1095,6 +1079,39 @@ void DetermineBasePaths(const char *exe)
 		homedir = (pw == nullptr) ? nullptr : stredup(pw->pw_dir);
 	}
 #endif
+	return homedir;
+}
+
+/**
+ * Determine the base (personal dir and game data dir) paths
+ * @param exe the path to the executable
+ */
+void DetermineBasePaths(const char *exe)
+{
+	char tmp[MAX_PATH];
+
+	char *homedir = GetHomeDir();
```

Comment:
"homedir" is leaked in various #if combinations

Reply:
Maybe bite the bullet and switch to `std::string` for this stuff?